pallets/file-system/src/utils.rs

```diff
@@ -1540,7 +1540,17 @@ where
             Error::<T>::NotABsp
         );
 
-        // TODO: charge SP for this action.
+        let bsp_account_id =
+            <T::Providers as shp_traits::ReadProvidersInterface>::get_owner_account(bsp_id)
+                .ok_or(Error::<T>::FailedToGetOwnerAccount)?;
```

Comment:
I would change the error for a `expect_or_err!` macro invocation since this should never error, as we are checking that the provider exists and is a BSP before, so it MUST have an owner account.